api/src/main/java/org/pmiops/workbench/api/CloudTaskEnvironmentsController.java

```diff
@@ -0,0 +1,162 @@
+package org.pmiops.workbench.api;
+
+import static org.pmiops.workbench.utils.LogFormatters.formatDurationPretty;
+
+import com.google.common.base.Stopwatch;
+import jakarta.inject.Provider;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.WorkbenchException;
+import org.pmiops.workbench.leonardo.LeonardoApiClient;
+import org.pmiops.workbench.leonardo.LeonardoStatusUtils;
+import org.pmiops.workbench.model.UserRole;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class CloudTaskEnvironmentsController implements CloudTaskEnvironmentsApiDelegate {
+  private static final Logger LOGGER =
+      Logger.getLogger(CloudTaskEnvironmentsController.class.getName());
+
+  private final LeonardoApiClient leonardoApiClient;
+  private final Provider<Stopwatch> stopwatchProvider;
+  private final WorkspaceService workspaceService;
+
+  @Autowired
+  public CloudTaskEnvironmentsController(
+      LeonardoApiClient leonardoApiClient,
+      Provider<Stopwatch> stopwatchProvider,
+      WorkspaceService workspaceService) {
+    this.leonardoApiClient = leonardoApiClient;
+    this.stopwatchProvider = stopwatchProvider;
+    this.workspaceService = workspaceService;
+  }
+
+  @Override
+  public ResponseEntity<Void> ctDeleteUnsharedWorkspaceEnvironments(
```

Comment:
"ct" here stands for Cloud Task, not Controlled Tier.  If that's confusing, I can find a different solution to the name collision